x-pack/plugin/core/src/main/resources/logs-mappings.json

```diff
@@ -2,6 +2,10 @@
   "template": {
     "mappings": {
       "properties": {
+        "@timestamp": {
+          "type": "date",
+          "ignore_malformed": false
```

Comment:
Is this needed for a data stream? How is this handled today in templates loaded by packages?

Reply:
It did not require any special handling so far. Now, since we set `ignore_malformed: true` at the [backing indices level](https://github.com/elastic/elasticsearch/pull/95329/files#diff-a073a100cddc5733e0a8581e8e0d5fb0b1c675e6f08633eb58b709a62f69a191R13), we need to make this exception explicitly for `@timestamp`, otherwise it fails [validation](https://github.com/elastic/elasticsearch/blob/da214475a5c9604f4f1a32557f319321eae4cdf1/server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java#L151-L154) (for a good reason).
In this specific case, we could only update the `logs-mapping`, but then I am not sure how well defined it is to have contradicting mappings within the [logs-template composition](https://github.com/elastic/elasticsearch/blob/a2c8e9c847e1f8bf8d47d2b7af2f4de7d877028d/x-pack/plugin/core/src/main/resources/logs-template.json#L6-L8) of all three sources. 
Besides, is there any harm in making it very explicit here? I assume it can only improve, as it will guard from a similar problem if other data stream mappings set `ignore_malformed` at the index level.

Reply:
Thanks for the details. I like that we make it explicit for timestamp.

Templates loaded by packages don't use this template, that is why @gsantoro works on the package part.